diesel/src/type_impls/tuples.rs

```diff
@@ -114,8 +114,11 @@ macro_rules! tuple_impls {
                 const HAS_STATIC_QUERY_ID: bool = $($T::HAS_STATIC_QUERY_ID &&)+ true;
             }
 
-            impl<$($T: Expression + NonAggregate),+> NonAggregate for ($($T,)+) {
-            }
+            const _: () = {
```

Comment:
Just out of curiosity: Why do we wrap this here in a const again?

Reply:
Any code in here gets repeated multiple times. We need a unique name to avoid conflicts. Since `concat_idents!` is unstable (and wouldn't work here anyway), we couldn't do this in `macro_rules!` until recently.

Reply:
The alternative would be to not use the derive here, but I don't think we would be able to generate the where clause of the impl from a `macro_rules!` macro

Reply:
Ah, yes. This is only required because every generated wrapping const has the same name, right? So this wouldn't be required with #2246 any more, right?

Reply:
It's still needed because the dummy struct we're using for the derive needs a unique name.